docs/api.rst

```diff
@@ -11486,3 +11486,174 @@ Handling user notes
+   :resjson object result: An array of objects representing custom assets entries.
+   :resjson int identifier: The unique identifier of the custom asset.
+   :resjson str name: The name of the custom asset.
+   :resjson str notes: The notes used as a description of the custom asset.
+   :resjson str custom_asset_type: The type/category of the custom asset.
+   :resjson int entries_found: The number of entries found for the current filter. Ignores pagination.
+   :resjson int entries_limit: The limit of entries. Constant value of -1.
+   :resjson int entries_total: The number of total entries ignoring all filters.
+
+   :statuscode 200: Custom assets were retrieved successfully.
+   :statuscode 409: No user is currently logged in.
+   :statuscode 500: Internal rotki error.
+
+.. http:delete:: /api/(version)/assets/custom
+
+   Doing a DELETE on this endpoint will delete a custom asset for the specified identifier.
+
+   **Example Request**:
+
+   .. http:example:: curl wget httpie python-requests
+
+      DELETE /api/1/assets/custom HTTP/1.1
+      Host: localhost:5042
+      Content-Type: application/json;charset=UTF-8
+
+      {"identifier": "aca42c5d-36f1-4a8b-af28-5aeb322748b5"}
+
+   :reqjson int identifier: The identifier of the custom asset you're trying to delete.
+
+   **Example Response**:
+
+   .. sourcecode:: http
+
+      HTTP/1.1 200 OK
+      Content-Type: application/zip
+
+      {"result": true, "message": ""}
+
+   :statuscode 200: Custom asset was deleted successfully.
+   :statuscode 400: Provided JSON is in some way malformed.
+   :statuscode 409: No user is currently logged in. No custom asset found. Check error message.
+   :statuscode 500: Internal rotki error.
+
+
+.. http:patch:: /api/(version)/assets/custom
+
+   Doing a PATCH on this endpoint will update the content of an already existing custom asset.
+
+   **Example Request**:
+
+   .. http:example:: curl wget httpie python-requests
+
+      PATCH /api/1/assets/custom HTTP/1.1
+      Host: localhost:5042
+      Content-Type: application/json;charset=UTF-8
+
+        {
+            "identifier": 2,
+            "name": "Hotel",
+            "notes": "Hotel from ma",
+            "custom_asset_type": "real estate"
+        }
+
+   :reqjson str identifier: The unique identifier of the custom asset.
+   :reqjson str name: The name of the custom asset.
+   :reqjson str[optional] notes: The notes that serve as a description of the custom asset. This is optional.
+   :reqjson str[optional] custom_asset_type: The type/category of the custom asset. This is optional
+
+   **Example Response**:
+
+   .. sourcecode:: http
+
+      HTTP/1.1 200 OK
+      Content-Type: application/zip
+
+      {"result": true, "message": ""}
+
+   :statuscode 200: Custom asset was updated successfully.
+   :statuscode 400: Provided JSON is in some way malformed.
+   :statuscode 409: No user is currently logged in. New custom asset name is in use. Custom asset does not exist.  Check error message.
```

Comment:
I still don't see a need to allow duplication(I might not be getting things clearly) but then if duplication is allowed and a user for instance has a custom asset's name as **House** and another custom asset's name as **House**, this will lead to confusion.